Add locations as owners for fixtures

esoergel opened the pull request:
http://manage.dimagi.com/default.asp?176166

Blocked on QA: http://manage.dimagi.com/default.asp?178049

esoergel committed:
Add dbaccessor for fixtures/ownership pattern

esoergel committed:
Add soft_assert to see if we can remove  edge case

esoergel committed:
Update fixture models to support location owners

esoergel committed:
Allow upload of fixtures with location owners

esoergel committed:
Include location owners in fixture download

esoergel committed:
Memoize location lookups

dimagimon commented on corehq/apps/fixtures/upload.py:
```diff
@@ -209,22 +210,38 @@ def get_workbook(file_or_filename):
-        group_memoizer.add_group(loc.case_sharing_group_object())
-        group_memoizer.add_group(loc.reporting_group_object())
+    locations = {}
+    def get_location(user_input):
```
E301 expected 1 blank line, found 0

esoergel committed:
Switch parsing of user input to manager method

esoergel commented on corehq/apps/importer/util.py:
```diff
@@ -455,11 +455,10 @@ def get_from_group(name):
-            try:
-                return SQLLocation.objects.get(domain=domain, **filter_).location_id
-            except SQLLocation.DoesNotExist:
-                pass
```
This isn't related to this PR at all - I was just moving this other thing I did which turns user input into a location into a model manager method so the logic is consistent.

esoergel committed:
Remove old locations-as-groups logic

Unneeded, now that locations are first-class fixture owners.  This is
largely a revert of 211fc4114

esoergel commented:
QA passed.  @proteusvacuum @emord
